TestingGrounds/TestingGrounds/HashTable.cpp

```diff
@@ -1,42 +1,40 @@
 #include "HashTable.hpp"
 
-template <uint32_t K, uint32_t V>
-HashTable<K, V>::HashTable():
-    capacity{ 20 },
+using namespace std;
+
+HashTable::HashTable():
     size{ 0 },
-    table(new HashNode<K, V>*[capacity])
+    table(new HashNode*[capacity])
 {
     for(int i = 0; i < capacity; i++)
         table[i] = nullptr;
 }
 
-template<uint32_t K, uint32_t V>
-HashTable<K, V>::~HashTable()
+HashTable::~HashTable()
 {
-    // We may not have reached full capacity by the time we delete
-    // the Hash Table
-    uint32_t cap = capacity > size? capacity : size;
-    for(uint8_t i = 0; i < cap; i++)
+    for(uint8_t i = 0; i < capacity; i++)
     {
-        delete table[i];
-        table[i] = nullptr;   
+        if(table[i] != nullptr)
+        {
+            delete table[i];
+            table[i] = nullptr;
+        }
     }
     delete table;
+    size = 0;
 }
 
-template<uint32_t K, uint32_t V>
-uint32_t HashTable<K, V>::hashFunction( void )
+uint32_t HashTable::hashFunction( uint32_t key )
 {
-    return V % size;
+    return key % capacity;
 }
 
 // Double hashing is one of the best forms of probing, does not yield
 // clusters, and is effective for resolving collisions. We do this 
 // 
-template<uint32_t K, uint32_t V>
-uint32_t HashTable<K, V>::doubleHashFunction( void )
+uint32_t HashTable::doubleHashFunction( uint32_t key )
 {
-    return calcPrime() - hashFunction(V);
+    return calcPrime() - hashFunction(key);
```

Comment:
If `calcPrime()` returns 0, you will return a wrapped number and all your memory will be yeeted into ES IV: Oblivion.

Reply:
Un yeeted